src/mwutils/mwmath.cpp

```diff
@@ -37,14 +34,14 @@ int MathUtils::ipow(int m, int e) {
  *	The norm of the matrix is obtained as:
  *	 \f$ ||M|| \lim_{n \rightarrow \infty} ||x_n||/||x_{n-1}||\f$
  */
-double MathUtils::matrixNorm2(const VectorXd &vector) {
-    int n = (int) sqrt(vector.size());
+double mwmath::matrixNorm2(const VectorXd &vector) {
```

Comment:
You might want to explore this https://eigen.tuxfamily.org/dox/group__TutorialReductionsVisitorsBroadcasting.html
Also, be aware that passing Eigen types as parameters might impact performance negatively, if not done correctly: https://eigen.tuxfamily.org/dox/TopicFunctionTakingEigenTypes.html

Reply:
I changed the matrix norms to the `lpNorm` routines of Eigen, but it turns out they require v.3.3. Is it possible to get this version on Travis without shipping our own  Eigen? Currently I get v3.2 when installing `libeigen3-dev` on Travis, while I get v3.3.4 on my laptop.

Reply:
For the Eigen types, I guess we are fine as long av we pass by `const` reference. This is done in the most performance critical routines at least.

Reply:
Add something like this to `.travis.yml`:
```
  cd $HOME/Downloads
  mkdir -p eigen
  curl -Ls http://bitbucket.org/eigen/eigen/get/${Eigen_VERSION}.tar.gz | tar -xz -C eigen --strip-components=1
  cd eigen
  cmake -H. -Bbuild_eigen -DCMAKE_INSTALL_PREFIX=$HOME/Deps/eigen &> /dev/null
  cmake --build build_eigen -- install &> /dev/null
  cd $TRAVIS_BUILD_DIR
  rm -rf $HOME/Downloads/eigen
```
and then when configuring the project add:
```
--cmake-options="-DEigen3_DIR: '$HOME/Deps/eigen/share/eigen3/cmake/"
```
The `find_package` for Eigen should be:
```
find_package(Eigen3 3.3 REQUIRED NO_MODULE)
```
and then you can do:
```
target_link_libraries(...
  PUBLIC
    Eigen3::Eigen
  )
```